dart/lib/src/debug_image_extractor.dart

```diff
@@ -0,0 +1,164 @@
+  }
+
+  bool _isHeaderStartLine(String line) {
+    return line.contains(_headerStartLine);
+  }
+
+  String? _extractBuildId(String line) {
+    final buildIdMatch = _buildIdRegex.firstMatch(line);
+    return buildIdMatch?.group(1);
+  }
+
+  String? _extractIsolateDsoBase(String line) {
+    final isolateMatch = _isolateDsoBaseLineRegex.firstMatch(line);
+    return isolateMatch?.group(1);
+  }
+}
+
+class _DebugInfo {
+  final String? buildId;
+  final String? isolateDsoBase;
+  final SentryOptions _options;
+
+  _DebugInfo(this.buildId, this.isolateDsoBase, this._options);
+
+  DebugImage? toDebugImage() {
+    if (buildId == null || isolateDsoBase == null) {
+      _options.logger(SentryLevel.warning,
+          'Cannot create DebugImage without buildId and isolateDsoBase.');
+      return null;
+    }
+
+    String type;
+    String? imageAddr;
+    String? debugId;
+    String? codeId;
+
+    final platform = _options.platformChecker.platform;
+
+    // Default values for all platforms
+    imageAddr = '0x$isolateDsoBase';
+
+    if (platform.isAndroid) {
+      type = 'elf';
+      debugId = _convertCodeIdToDebugId(buildId!);
+      codeId = buildId;
+    } else if (platform.isIOS || platform.isMacOS) {
+      type = 'macho';
+      debugId = _formatHexToUuid(buildId!);
+      // `codeId` is not needed for iOS/MacOS.
+    } else {
+      _options.logger(
+        SentryLevel.warning,
+        'Unsupported platform for creating Dart debug images.',
+      );
+      return null;
+    }
+    return DebugImage(
+      type: type,
+      imageAddr: imageAddr,
+      debugId: debugId,
+      codeId: codeId,
+    );
+  }
+
+  // Debug identifier is the little-endian UUID representation of the first 16-bytes of
+  // the build ID on ELF images.
+  String? _convertCodeIdToDebugId(String codeId) {
+    codeId = codeId.replaceAll(' ', '');
+    if (codeId.length < 32) {
+      _options.logger(SentryLevel.warning,
+          'Code ID must be at least 32 hexadecimal characters long');
+      return null;
+    }
+
+    final first16Bytes = codeId.substring(0, 32);
+    final byteData = UuidParsing.parseHexToBytes(first16Bytes);
+
+    if (byteData.isEmpty) {
+      _options.logger(
+          SentryLevel.warning, 'Failed to convert code ID to debug ID');
+      return null;
+    }
+
+    return bigToLittleEndianUuid(UuidValue.fromByteList(byteData).uuid);
+  }
+
+  String bigToLittleEndianUuid(String bigEndianUuid) {
+    // Remove hyphens and convert to a byte array
+    final byteArray =
+        Uuid.parse(bigEndianUuid, validationMode: ValidationMode.nonStrict);
+
+    // Reverse the necessary sections according to the UUID fields
+    final reversedByteArray = Uint8List.fromList([
+      ...byteArray.sublist(0, 4).reversed,
+      ...byteArray.sublist(4, 6).reversed,
+      ...byteArray.sublist(6, 8).reversed,
+      ...byteArray.sublist(8, 10),
+      ...byteArray.sublist(10),
+    ]);
+
+    return Uuid.unparse(reversedByteArray);
+  }
```

Comment:
not sure how performant this is

Reply:
hopefully we call it only once, so it shouldn't be a problem

Reply:
yup we do